src/components/common/UserProfileEdit/AdvancedSettingsRow.tsx

```diff
@@ -0,0 +1,190 @@
+  metatransactionsDisabled,
+}: MetatransactionsDisabledProps) =>
+  metatransactionsDisabled ? (
+    <div className={stylesAdvance.metaDesc}>
+      <FormattedMessage {...MSG.metaDescGlobalOff} />
+    </div>
+  ) : null;
+
+const CustomEndpointInput = () => {
+  const {
+    watch,
+    formState: { isValid },
+  } = useFormContext();
+  const decentralizedMode: boolean = watch(SlotKey.DecentralizedMode);
+  const customRpc: string = watch(SlotKey.CustomRPC);
+
+  return (
+    <>
+      <Input
+        label={MSG.labelRPC}
+        appearance={{ colorSchema: 'grey' }}
+        name={SlotKey.CustomRPC}
+        disabled={!decentralizedMode}
+        formattingOptions={noSpaces}
+        value={customRpc}
+      />
+      <div className={stylesAdvance.validateButtonContainer}>
+        <Button
+          text={MSG.validate}
+          disabled={!decentralizedMode || !isValid}
+          // @TODO: validation to be added later
+        />
+      </div>
+    </>
+  );
+};
+
+export const getAdvancedSettingsRows = (
+  metatransactionsAvailable: boolean,
+): Omit<SettingsRowProps, 'toggleDisabled'>[] => [
+  {
+    name: SlotKey.Metatransactions,
+    paragraphText: MSG.metaDescription,
+    toggleLabel: MSG.labelMetaTx,
+    tooltipText: MSG.metaTooltip,
+    tooltipTextValues: { br: <br /> },
+    extra: (
+      <MetatransactionsDisabled
+        metatransactionsDisabled={!metatransactionsAvailable}
+      />
+    ),
+  },
+  {
+    name: SlotKey.DecentralizedMode,
+    paragraphText: MSG.endpointsDescription,
+    toggleLabel: MSG.customEndpoints,
+    tooltipText: MSG.RPCTooltip,
+    extra: <CustomEndpointInput />,
+  },
+];
+
+interface SettingsRowProps {
+  extra: ReactElement;
+  paragraphText: MessageDescriptor;
+  name: string;
+  toggleDisabled: boolean;
+  toggleLabel: MessageDescriptor;
+  tooltipText: MessageDescriptor;
+  tooltipTextValues?: UniversalMessageValues;
+}
+
+const AdvancedSettingsRow = ({
+  extra,
+  name,
+  paragraphText,
+  toggleDisabled: disabled,
+  toggleLabel,
+  tooltipText,
+  tooltipTextValues,
+}: SettingsRowProps) => {
+  const { trigger, watch } = useFormContext();
+  const value: boolean = watch(name);
+  return (
+    <>
+      <div className={stylesAdvance.toggleContainer}>
+        <Toggle
+          label={toggleLabel}
+          labelValues={{ isOn: value }}
+          name={name}
+          disabled={disabled}
+          onChange={() => trigger()}
+        />
+        <QuestionMarkTooltip
+          tooltipText={tooltipText}
+          tooltipTextValues={tooltipTextValues}
+          className={stylesAdvance.tooltipContainer}
+          tooltipClassName={stylesAdvance.tooltipContent}
+          tooltipPopperOptions={{
+            placement: 'right',
+          }}
+        />
+      </div>
+      <p className={stylesAdvance.descriptions}>
+        <FormattedMessage {...paragraphText} />
+      </p>
+      <div className={stylesAdvance.settingsRowExtra}>{extra}</div>
+      <hr />
```

Comment:
Not really a big fan of `hr` elements, but then again this might be just a personal nitpick rather than an objective complaint